File: backend/src/HerPace.Core/Entities/ResearchStudy.cs
namespace HerPace.Core.Entities;

/// <summary>
/// A peer-reviewed research study powering evidence-based training recommendations.
/// </summary>
public class ResearchStudy
{
    public int Id { get; set; }
    public string ResearchTopic { get; set; } = string.Empty;
    public string Citation { get; set; } = string.Empty;
    public string? Doi { get; set; }
    public string StudyDesign { get; set; } = string.Empty;
    public string? SampleSize { get; set; }
    public int? PublicationYear { get; set; }
    public string KeyFindings { get; set; } = string.Empty;
    public string EvidenceTier { get; set; } = string.Empty;
    public string TopicCategory { get; set; } = string.Empty;

    // Navigation property
    public ICollection<PhaseStudyMapping> PhaseStudyMappings { get; set; } = new List<PhaseStudyMapping>();
}

File: android/app/src/main/java/com/herpace/domain/model/ResearchStudy.kt
package com.herpace.domain.model

data class ResearchStudySummary(
    val id: Int,
    val researchTopic: String,
    val citation: String,
    val publicationYear: Int?,
    val evidenceTier: String,
    val topicCategory: String
)

data class ResearchStudyDetail(
    val id: Int,
    val researchTopic: String,
    val citation: String,
    val doi: String?,
    val studyDesign: String,
    val sampleSize: String?,
    val publicationYear: Int?,
    val keyFindings: String,
    val evidenceTier: String,
    val topicCategory: String,
    val phaseRelevance: List<PhaseRelevance>
)

data class PhaseRelevance(
    val phase: Int,
    val phaseName: String,
    val relevanceSummary: String
)
